src/plugins/workspace/server/saved_objects/saved_objects_wrapper_for_check_workspace_conflict.ts

```diff
@@ -0,0 +1,317 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import Boom from '@hapi/boom';
+import {
+  SavedObject,
+  SavedObjectsBaseOptions,
+  SavedObjectsBulkCreateObject,
+  SavedObjectsBulkResponse,
+  SavedObjectsClientWrapperFactory,
+  SavedObjectsCreateOptions,
+  SavedObjectsErrorHelpers,
+  SavedObjectsSerializer,
+  SavedObjectsCheckConflictsObject,
+  SavedObjectsCheckConflictsResponse,
+} from '../../../../core/server';
+
+const errorContent = (error: Boom.Boom) => error.output.payload;
+
+const filterWorkspacesAccordingToSourceWorkspaces = (
+  targetWorkspaces?: string[],
+  baseWorkspaces?: string[]
+): string[] => targetWorkspaces?.filter((item) => !baseWorkspaces?.includes(item)) || [];
+
+export class WorkspaceConflictSavedObjectsClientWrapper {
+  private _serializer?: SavedObjectsSerializer;
+  public setSerializer(serializer: SavedObjectsSerializer) {
+    this._serializer = serializer;
+  }
+  private getRawId(props: { namespace?: string; id: string; type: string }) {
+    return (
+      this._serializer?.generateRawId(props.namespace, props.type, props.id) ||
+      `${props.type}:${props.id}`
+    );
+  }
+  public wrapperFactory: SavedObjectsClientWrapperFactory = (wrapperOptions) => {
+    const createWithWorkspaceConflictCheck = async <T = unknown>(
+      type: string,
+      attributes: T,
+      options: SavedObjectsCreateOptions = {}
+    ) => {
+      const { workspaces, id, overwrite } = options;
+      let savedObjectWorkspaces = options?.workspaces;
+
+      /**
+       * Check if overwrite with id
+       * If so, need to reserve the workspace params
+       */
+      if (id && overwrite) {
+        let currentItem;
+        try {
+          currentItem = await wrapperOptions.client.get(type, id);
+        } catch (e) {
```

Comment:
is it possible to handle specific error here? for example, e.response && e.response.status === 404, so we can avoid unintentionally swallowing other types of errors.

Reply:
Sure, let me optimize this part.